File: OCT2023 GFG/Height of Binary Tree.c
int max(int a, int b) {
    return (a > b) ? a : b;
}

//Function to find the height of a binary tree.
int height(struct Node* node)
{
    //code here
    if (node == NULL) {
        return 0;
    } else {
        int leftHeight = height(node->left);
        int rightHeight = height(node->right);

        return 1 + max(leftHeight, rightHeight);
    }
}

File: FEB 2025/Height of Binary Tree.py
class Solution:
    #Function to find the height of a binary tree.
    def height(self, root):
        # code here
        if root is None:
            return -1
        
        # Recursively compute the height of the left and right subtrees
        left_height = self.height(root.left)
        right_height = self.height(root.right)
        
        # The height of the tree is the greater height of its two subtrees, plus one edge for the current node.
        return max(left_height, right_height) + 1
